components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java

```diff
@@ -117,16 +122,19 @@ private String buildResponse(Map<String, Certificate> certificates)
         // If we read different algorithms from identity.xml then put them in a list.
         List<JWSAlgorithm> diffAlgorithms = findDifferentAlgorithms(accessTokenSignAlgorithm, config);
         // Create JWKS for different algorithms using new KeyID creation method.
-        for (Map.Entry certificateWithAlias : certificates.entrySet()) {
+        for (CertificateInfo certInfo : certInfos) {
             for (JWSAlgorithm algorithm : diffAlgorithms) {
-                Certificate cert = (Certificate) certificateWithAlias.getValue();
-                String alias = (String) certificateWithAlias.getKey();
-                RSAPublicKey publicKey = (RSAPublicKey) cert.getPublicKey();
-                RSAKey.Builder jwk = new RSAKey.Builder(publicKey);
+                String alias = certInfo.getCertificateAlias();
+                X509Certificate cert = (X509Certificate) certInfo.getCertificate();
+                Certificate[] certChain = certInfo.getCertificateChain();
+                List<Base64> certList = generateEncodedCertList(certChain, alias);
```

Comment:
```suggestion
                List<Base64> encodedCertList = generateEncodedCertList(certChain, alias);
```
Having multiple cert list may lead for confusion